pkg/list/values/extract.go

```
@@ -0,0 +1,249 @@
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/list/errors"
+	"github.com/cloudposse/atmos/pkg/utils"
+)
+
+// handleSpecialComponent processes special components like settings and metadata.
+func handleSpecialComponent(stack map[string]interface{}, component string) (map[string]interface{}, bool) {
```

golangci-lint (1.62.2)

cognitive complexity 26 of func `handleSpecialComponent` is high (> 20)

(gocognit)